docs/utils/masked.rst

```diff
@@ -0,0 +1,136 @@
+`~astropy.utils.masked.Masked.unmasked`, or, if you need to control what
+should be substituted for any masked values, with
+:meth:`~astropy.utils.masked.Masked.unmask`::
+
+  >>> mq.unmasked
+  <Quantity [1., 2., 3.] m>
+  >>> mq.unmask(-75*u.cm)
+  <Quantity [ 1.  ,  2.  , -0.75] m>
+
+For reductions such as sums, the mask propagates as if the sum was
+done directly::
+
+  >>> ma = Masked([[0., 1.], [2., 3.]], mask=[[False, True], [False, False]])
+  >>> ma.sum(-1)
+  MaskedNDArray([———, 5.0])
+  >>> ma.sum()
+  MaskedNDArray(———)
+
+You might wonder why masked elements are propagated, instead of just being
+skipped (as is done in `~numpy.ma.MaskedArray`; see :ref:`below
+<utils-masked-vs-numpy-maskedarray>`).  The rationale is that this leaves a
+sum which is generally not useful unless one knows the number of masked
+elements.  In contrast, for sample properties such as the mean, for which the
+number of elements are counted, it seems natural to simply omit the masked
+elements from the calculation::
+
+  >> ma.mean(-1)
+  MaskedNDArray([0.0, 2.5])
+
+
+.. _utils-masked-vs-numpy-maskedarray:
+
+Differences from `numpy.ma.MaskedArray`
+=======================================
+
+|Masked| differs from `~numpy.ma.MaskedArray` in a number of ways.  In usage,
+a major difference is that most operations act on the masked values, i.e., no
+effort is made to preserve values.  For instance, compare::
+
+  >>> np_ma = np.ma.MaskedArray([1., 2., 3.], mask=[False, True, False])
+  >>> (np_ma + 1).data
+  array([2., 2., 4.])
+  >>> (Masked(np_ma) + 1).unmasked
+  array([2., 3., 4.])
+
+The main reason for this decision is that for some masked subclasses, like
+masked |Quantity|, keeping the original value makes no sense (e.g., consider
+dividing a length by a time: if the unit of a masked quantity is changing, why
+should its value not change?).  But it also helps to keep the implementation
+considerably simpler, as the |Masked| class now primarily has to deal with
+propagating the mask rather than deciding what to do with values.
+
+A second difference is that for reductions, the mask propagates as it would
+have if the operations were done on the individual elements::
+
+  >>> np_ma.prod()
+  3.0
+  >>> np_ma[0] * np_ma[1] * np_ma[2]
+  masked
+  >>> Masked(np_ma).prod()
+  MaskedNDArray(———)
+
+The rationale for this becomes clear again by thinking about subclasses like a
+masked |Quantity|.  For instance, if you had an array of lengths that
+represented width, height, and depth as their last axis, the dimension of the
+product over the last axis would depend on how many items were masked, which
+|Quantity| could not represent (and would be rather surprising!).  As noted
+above, however, masked elements are skipped for operations for which this is
+well defined, such as for getting the mean and other sample properties such as
+the variance and standard deviation.
+
+A third difference is more conceptual.  For `~numpy.ma.MaskedArray`, the
```

Comment:
I don't totally understand this paragraph (and I actually wrote an ndarray sublcass once, albeit long ago). I think a more useful discussion for most users to just note that while `mq = np.ma.MaskedArray([1, 2]*u.m)` appears to do something useful on first glance, it actually just does not work. E.g.
```
In [40]: np.ma.array([1,2]*u.m) / u.s                                           
Out[40]: <Quantity [1., 2.] 1 / s>
```
So it is not just a conceptual break, it is paradigm change that allows usefully wrapping ndarray-subclasses.

You might also note here that the astropy `Column` (which is an ndarray subclass) can easily be wrapped with `Masked` and it all *just works*. (But that we choose not to do so because of the behavior changes).

Reply:
Yes to examples!  I added some and also mention `Column`.